hudi-cli/src/main/java/org/apache/hudi/cli/commands/BootstrapCommand.java

```diff
@@ -0,0 +1,185 @@
+    UtilHelpers.validateAndAddProperties(configs, sparkLauncher);
+    Process process = sparkLauncher.launch();
+    InputStreamConsumer.captureOutput(process);
+    int exitCode = process.waitFor();
+    if (exitCode != 0) {
+      return "Failed to bootstrap source data to Hudi dataset";
+    }
+    return "Bootstrapped source data as Hudi dataset";
+  }
+
+  @CliCommand(value = "bootstrap index showMapping", help = "Show bootstrap index mapping")
+  public String showBootstrapIndexMapping(
+      @CliOption(key = {"partitionPath"}, unspecifiedDefaultValue = "", help = "A valid paritition path") String partitionPath,
+      @CliOption(key = {"fileIds"}, unspecifiedDefaultValue = "", help = "Valid fileIds split by comma") String fileIds,
+      @CliOption(key = {"limit"}, unspecifiedDefaultValue = "-1", help = "Limit rows to be displayed") Integer limit,
+      @CliOption(key = {"sortBy"}, unspecifiedDefaultValue = "", help = "Sorting Field") final String sortByField,
+      @CliOption(key = {"desc"}, unspecifiedDefaultValue = "false", help = "Ordering") final boolean descending,
+      @CliOption(key = {"headeronly"}, unspecifiedDefaultValue = "false", help = "Print Header Only")
+      final boolean headerOnly) {
+
+    if (partitionPath.isEmpty() && !fileIds.isEmpty()) {
+      throw new IllegalStateException("When passing fileIds, partitionPath is mandatory");
+    }
+    BootstrapIndex.IndexReader indexReader = createBootstrapIndexReader();
+    List<String> indexedPartitions = indexReader.getIndexedPartitionPaths();
+
+    if (!partitionPath.isEmpty() && !indexedPartitions.contains(partitionPath)) {
+      return partitionPath + " is not an valid indexed partition";
+    }
+
+    List<BootstrapFileMapping> mappingList = new ArrayList<>();
+    if (!fileIds.isEmpty()) {
+      List<HoodieFileGroupId> fileGroupIds = Arrays.stream(fileIds.split(","))
+          .map(fileId -> new HoodieFileGroupId(partitionPath, fileId)).collect(Collectors.toList());
+      mappingList.addAll(indexReader.getSourceFileMappingForFileIds(fileGroupIds).values());
+    } else if (!partitionPath.isEmpty()) {
+      mappingList.addAll(indexReader.getSourceFileMappingForPartition(partitionPath));
+    } else {
+      for (String part : indexedPartitions) {
+        mappingList.addAll(indexReader.getSourceFileMappingForPartition(part));
+      }
+    }
+
+    final List<Comparable[]> rows = convertBootstrapSourceFileMapping(mappingList);
+    final TableHeader header = new TableHeader()
+        .addTableHeaderField("Hudi Partition")
+        .addTableHeaderField("FileId")
+        .addTableHeaderField("Source File Base Path")
+        .addTableHeaderField("Source File Parition")
+        .addTableHeaderField("Source File Path");
+
+    return HoodiePrintHelper.print(header, new HashMap<>(), sortByField, descending,
+        limit, headerOnly, rows);
+  }
+
+  @CliCommand(value = "bootstrap index showPartitions", help = "Show bootstrap indexed partitions")
```

Comment:
TO preserve uniformity with other CLIs, I renamed the CLI to be "bootstrap index showpartitions"